ruma-federation-api/src/event/get_event/v1.rs

```diff
@@ -0,0 +1,58 @@
+//! [GET /_matrix/federation/v1/event/{eventId}](https://matrix.org/docs/spec/server_server/r0.1.4#get-matrix-federation-v1-event-eventid)
+
+use js_int::UInt;
+use ruma_api::ruma_api;
+use ruma_events::pdu::Pdu;
+use ruma_identifiers::{EventId, ServerNameBox};
+
+ruma_api! {
+    metadata: {
+        description: "Retrieves a single event.",
+        method: GET,
+        name: "get_event",
+        path: "/_matrix/federation/v1/event/:event_id",
+        rate_limited: false,
+        requires_authentication: true,
+    }
+
+    #[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
+    request: {
+        /// The event ID to get.
+        #[ruma_api(path)]
+        event_id: &'a EventId,
+    }
+
+    #[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
+    response: {
+        /// The `server_name` of the homeserver sending this transaction.
+        origin: ServerNameBox,
+
+        /// POSIX timestamp in milliseconds on originating homeserver when this
+        /// transaction started.
+        origin_server_ts: UInt,
+
+        /// A single PDU.
+        ///
+        /// Note that events have a different format depending on the room
+        /// version - check the [room version specification] for precise event
+        /// formats.
+        ///
+        /// [room version specification]: https://matrix.org/docs/spec/index.html#room-versions
+        pdus: Vec<Pdu>
+    }
+}
+
+impl<'a> Request<'a> {
+    /// Creates a new `Request` with the given event id.
+    pub fn new(event_id: &'a EventId) -> Self {
+        Self { event_id }
+    }
+}
+
+impl Response {
+    /// Creates a new `Response` with the given origin, server timestamp, and
+    /// list containing a single `Pdu`.
+    pub fn new(origin: ServerNameBox, origin_server_ts: UInt, pdus: Vec<Pdu>) -> Self {
```

Comment:
Wasn't sure whether to have `pdus` be a single `Pdu` to match what is presumably intended or a `Vec` to match what is sent on the wire.

Reply:
Yeah, that's weird. I think that requires spec clarification. Do you know if there is an open spec issue about that? If not, it would be good to create one to resolve the mismatch between the type and the description.

Reply:
I looked and didn't see any open issues. Created one at https://github.com/matrix-org/matrix-doc/issues/2768

Reply:
Ok looks like this should be vec to match transaction format.

Reply:
Idk what the 'transaction' format is in this context, but if it is an error for there to be multiple values in the JSON array, we should really encode this as just a `Pdu` with custom (de)serialization, sth. like `serde(with = "ruma_serde::single_element_seq")`. That said, I'd be fine merging with `Vec<Pdu>` and a FIXME comment that it can really just be one element.

Reply:
`ruma_serde::single_element_seq` sounds good I'll update that 👍 (the response is a single pdu transaction defined here: https://github.com/matrix-org/matrix-doc/blob/master/api/server-server/definitions/single_pdu_transaction.yaml#L21)